### PR TITLE
Added local dev server instructions and scritps

### PR DESCRIPTION
**Pull request by shinglyu**
I moved the scripts to the root level to avoid publishing them to the public accidentally.

### CODE REVIEW

File: dev-router.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// 2. Visit http://localhost:8000
+
+if (file_exists(__DIR__ . '/webroot' . $_SERVER['REQUEST_URI'])) {
+  // Serve the requested resource as-is.
+  return false;
+}
+
+if ($_SERVER['REQUEST_URI'] === '/static/scripts/combined.js') {
+  // Map combined.js to PHP
+  include('components/combine-scripts.php');
```
Comment (kyoshino) [inline dev-router.php]: `combine-scripts.php` cannot be loaded somehow. `webroot/components/combine-scripts.php` worked for me.

File: dev-router.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+} else {
+  // Handle everything else
+  include('index.php');
```
Comment (kyoshino) [inline dev-router.php]: Same here, `webroot/index.php` worked.

**Comment by kyoshino** [comment]
One more thing: `output-script-elements.inc.php` cannot be loaded when the URL is `/home/inbox` or something else. Can you change the one line in `index.php` like this:
```php
<?php include_once($_SERVER['DOCUMENT_ROOT'] . '/components/output-script-elements.inc.php'); ?>
```

File: dev-router.php (hunk 1 above)
Comment (shinglyu) [inline dev-router.php]: It worked for me this morning, I guess the difference is the `-t webroot` parameter I used in `run_dev_server.sh`. I'll double check when I get to office tomorrow.

**Comment by shinglyu** [comment]
Sure, I'll fix that too.

File: dev-router.php (hunk 1 above)
Comment (shinglyu) [inline dev-router.php]: Oops, I guess it's the the browsers cache that made it work. Fixed as you suggested.

**Comment by kyoshino** [comment]
Thanks, it's working pretty well! Sorry but one more thing: looks like web fonts are served as `text/html` because the `woff2` file extension is not supported by the [dev server](http://php.net/manual/en/features.commandline.webserver.php). Especially Font Awesome is unreadable in that case so would you mind adding this to your router?
```php
if (strpos($_SERVER['REQUEST_URI'], '.woff2') !== false) {
  // Serve WOFF2 files properly
  header('Content-Type: application/font-woff2');
  return false;
}
```
I think that's everything so far. `.htaccess` has Content Security Policy and some other extras but those are probably not essential on the dev server.

**Comment by kyoshino** [comment]
Oops, the app now doesn't work on Apache. :crying_cat_face: `.htaccess` needs this, maybe in the `Options` section:
```conf
DirectoryIndex index.php
```

**Comment by shinglyu** [comment]
Oh, I though `index.php` is also on the possible homepage filename list of Apache.

**Comment by kyoshino** [comment]
Yeah, it could be added to `httpd.conf` file but the option in `.htaccess` is helpful for those who forgot to have it, like me :stuck_out_tongue_winking_eye:

**Comment by kyoshino** [comment]
I'm merging this now. Thanks again @shinglyu!

**Comment by shinglyu** [comment]
@kyoshino : Thanks! If you need me to squash the commmits into one before merge, please let me know.

**Comment by kyoshino** [comment]
Ah I forgot to say that ;-)

**Pull request merged**